ea-factories/src/factories/balance.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  ExecuteFactory,
+  Config,
+  DataResponse,
+  Account,
+  SequenceResponseData,
+} from '@chainlink/types'
+import objectPath from 'object-path'
+import { Validator, AdapterError } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+
+const DEFAULT_DATA_PATH = 'addresses'
+const DEFAULT_CONFIRMATIONS = 6
+
+const WARNING_UNSUPPORTED_PARAMS = 'No Operation: this provider does not support'
+const ERROR_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+export function addresses(accounts: Account[]) {
+  return accounts.reduce<string[]>((accumulator, current) => {
+    if (!current.warning) accumulator.push(current.address)
+    return accumulator
+  }, [])
+}
+
+export type IsSupported = (coin: string, chain: string) => boolean
+export type BalancesResponse = DataResponse<Account[], any>
+export type GetBalance = (account: Account, config: BalanceConfig) => Promise<BalancesResponse>
+export type GetBalances = (accounts: Account[], config: BalanceConfig) => Promise<BalancesResponse>
+
+export type BalanceConfig = Config & {
+  confirmations?: number
+  shouldOverwrite?: boolean
+  verbose?: boolean
+  isSupported: IsSupported
+  getBalance?: GetBalance
+  getBalances?: GetBalances
+}
+
+const requireArray = (jobRunID: string, dataPath: string, data: any) => {
+  const inputData = <Account[]>objectPath.get(data, dataPath)
+
+  // Check if input data is valid
+  if (!inputData || !Array.isArray(inputData) || inputData.length === 0)
+    throw new AdapterError({
+      jobRunID,
+      message: `Input, at '${dataPath}' path, must be a non-empty array.`,
+      statusCode: 400,
+    })
+
+  return inputData
+}
+
+const toValidAccount = (jobRunID: string, account: Account, config: BalanceConfig): Account => {
+  // Is it possible to process?
+  if (!account.address)
+    throw new AdapterError({
+      jobRunID,
+      message: ERROR_MISSING_ADDRESS,
+      statusCode: 400,
+    })
+
+  // Defaults
+  if (!account.chain) account.chain = 'mainnet'
+  if (!account.coin) account.coin = 'btc'
+
+  // Should we process?
+  if (!config.shouldOverwrite && typeof account.balance === 'number') return account
```

Comment:
This flag is not documented and I am having trouble figuring out how it should be used and why it's useful?

Reply:
So this should probably be brought to a separate issue as well (and possibly removed here for now).

This flag _specifically_ for composite adapters knowing whether they should write on top of previous results or if they should skip when a result is already filled. Composite adapter specific configs haven't been implemented yet and might need some thought.

Reply:
Let's remove this for now, as we are not using it at the moment. You can add your thoughts and ideas on this [here](https://github.com/smartcontractkit/external-adapters-js/issues/175). Thanks!

Reply:
Removed for now :+1: